services/QuillLMS/app/services/provider_classroom_users_updater.rb

```diff
@@ -0,0 +1,48 @@
+class ProviderClassroomUsersUpdater
+  attr_reader :provider_classroom_id, :provider_user_ids
+
+  def initialize(provider_classroom_id, provider_user_ids)
+    @provider_classroom_id = provider_classroom_id
+    @provider_user_ids = provider_user_ids
+  end
+
+  def run
+    update_deleted_to_active
+    update_active_to_deleted
+    create_new_active
+  end
+
+  private def provider_classroom_user_class
+    ProviderClassroomUser
+  end
+
+  private def update_deleted_to_active
+    provider_classroom_user_class
+      .deleted
+      .where(provider_classroom_id: provider_classroom_id)
+      .where(provider_user_id: provider_user_ids)
+      .update_all(deleted_at: nil)
+  end
+
+  private def update_active_to_deleted
+    provider_classroom_user_class
+      .active
+      .where(provider_classroom_id: provider_classroom_id)
+      .where.not(provider_user_id: provider_user_ids)
+      .update_all(deleted_at: Time.now)
+  end
+
+  private def create_new_active
+    provider_classroom_user_class.create_list(provider_classroom_id, new_provider_user_ids)
+  end
+
+  private def existing_provider_user_ids
+    provider_classroom_user_class
```

Comment:
Just noting that this is a query that wouldn't use the compound index in the other PR.

Reply:
Resolved via earlier [PR](https://github.com/empirical-org/Empirical-Core/pull/8159/files#diff-f146075b9b5ec6e8211bb8a474bb0e03c1e1a6c111a85954ebec07b95a8a3249R13)